iosApp/widget/CommuteConfigurationAppIntent.swift

```diff
@@ -0,0 +1,218 @@
+        When(\.$autoReverse, .equalTo, true) {
+            Summary {
+                \.$originStation
+                \.$destinationStation
+                \.$timeDisplay
+                \.$autoReverse
+                \.$reverseDays
+                \.$reverseStartHour
+                \.$reverseEndHour
+                \.$showLastRefreshedTime
+            }
+        } otherwise: {
+            Summary {
+                \.$originStation
+                \.$destinationStation
+                \.$timeDisplay
+                \.$autoReverse
+                \.$showLastRefreshedTime
+            }
+        }
+    }
+    
+    @Parameter(title: "Origin Station", default: .exp)
+    var originStation: StationChoice
+
+    @Parameter(title: "Destination Station", default: .wtc)
+    var destinationStation: StationChoice
+    
+    @Parameter(title: "Time", default: TimeDisplay.relative)
+    var timeDisplay: TimeDisplay
+    
+    @Parameter(title: "Auto-reverse for commute", default: false)
+    var autoReverse: Bool
+    
+    @Parameter(
+        title: "Reverse from",
+        default: .pm12,
+        requestValueDialog: IntentDialog("When should stations reverse?")
+    )
+    var reverseStartHour: HourOfDay?
+    
+    @Parameter(
+        title: "Reverse until",
+        default: .am3,
+        requestValueDialog: IntentDialog("When should reversal end?")
+    )
+    var reverseEndHour: HourOfDay?
+
+        @Parameter(
+        title: "Reverse on days",
+        default: [Weekday.monday, Weekday.tuesday, Weekday.wednesday, Weekday.thursday, Weekday.friday]
+    )
+    var reverseDays: [Weekday]
+
+    @Parameter(title: "Show last refreshed time", default: false)
+    var showLastRefreshedTime: Bool
+    
+    init() {
+        self.originStation = .closest
+        self.destinationStation = .wtc
+        self.timeDisplay = .relative
+        self.autoReverse = false
+        self.reverseDays = Weekday.weekdays
+        self.reverseStartHour = .pm12
+        self.reverseEndHour = .am3
+    }
+    
+    init(
+        originStation: StationChoice = .exp,
+        destinationStation: StationChoice = .wtc,
+        timeDisplay: TimeDisplay = .relative,
+        autoReverse: Bool = false,
+        showLastRefreshedTime: Bool = false,
+        reverseDays: [Weekday] = Weekday.weekdays,
+        reverseStartHour: HourOfDay? = nil,
+        reverseEndHour: HourOfDay? = nil
+    ) {
+        self.originStation = originStation
+        self.destinationStation = destinationStation
+        self.timeDisplay = timeDisplay
+        self.autoReverse = autoReverse
+        self.showLastRefreshedTime = showLastRefreshedTime
+        self.reverseDays = reverseDays
+        self.reverseStartHour = reverseStartHour
+        self.reverseEndHour = reverseEndHour
+    }
+    
+    // Helper function to determine if stations should be reversed based on current time and day
+    func shouldReverseStations() -> Bool {
```

Comment:
This logic is also present in `SchedulePathApi.isActiveAt`. We should unify the logic since we're using an equivalent schedule format here